weaviate/graphql/aggregate.go

```diff
@@ -130,20 +171,14 @@ func (ab *AggregateBuilder) createFilterClause() string {
 		if ab.withWhereFilter != nil {
 			filters = append(filters, ab.withWhereFilter.String())
 		}
-		if ab.withNearTextFilter != nil {
-			filters = append(filters, ab.withNearTextFilter.build())
-		}
-		if ab.withNearVectorFilter != nil {
-			filters = append(filters, ab.withNearVectorFilter.build())
-		}
-		if ab.withNearObjectFilter != nil {
-			filters = append(filters, ab.withNearObjectFilter.build())
-		}
-		if ab.withAsk != nil {
-			filters = append(filters, ab.withAsk.build())
-		}
-		if ab.withNearImage != nil {
-			filters = append(filters, ab.withNearImage.build())
+		for _, b := range []argumentBuilder{
+			ab.withAsk, ab.withNearTextFilter, ab.withNearObjectFilter, ab.withNearVectorFilter, ab.withNearImage,
+			ab.withNearAudio, ab.withNearVideo, ab.withNearDepth, ab.withNearThermal, ab.withNearImu,
+		} {
+			bVal := reflect.ValueOf(b)
+			if bVal.Kind() == reflect.Ptr && !bVal.IsNil() {
+				filters = append(filters, b.build())
+			}
```

Comment:
wouldn't it be better to make a generic function for this? readability of this is a bit a poor

Reply:
didn't find simpler and working implementation